MovingBlocks/Terasology
# Player is shown the respawn screen, if he joins a game where his character is dead.

Until now, when the player joins a game, and the `restorePlayer` function is called, it checks whether it has the AliveComponent, and if not, then it attaches one. 
So, this allowed the dead players to join the game with 0 health, which doesnt seem logical. So, instead with this change, the players who are dead are shown the respawn screen, and only upon clicking the `respawn` button does the player get back in the game.



### Contains

A small logical fix to the dead-players rejoining the game.

### How to test

- Generate a new game.
- Type in `hjump` command, and jump. Upon landing, you will be shown the DeathScreen.
<b> Make sure you dont have `softLanding` cheat enabled. </b>
- Now, don't respawn, instead directly 'Exit Terasology'.
- Run the build again, and join the same game.
- You'll be shown the respawn screen before spawning into the world.

### Outstanding before merging
- Nothing as of now.
